### PR TITLE
Improve performance

### PR DESCRIPTION
**Pull request by jim-v**
This PR adds the following functionality:

   The raw camera pose relative to the map frame is published to allow an external program to perform global optimization of the generated map.

   A parameter, map_read_only prevents the map from being updated (default false).

   A script, fit_plane.py has been added to the scripts directory.  The script fits a plane to the fiducials in the map file, so that the quality of the map can be assessed.

This is part of ongoing work on #62, which has become more urgent due to #114.

### CODE REVIEW

File: fiducial_slam/src/map.cpp [hunk 1]
```diff
@@ -177,8 +177,11 @@ Map::Map(ros::NodeHandle &nh) : tfBuffer(ros::Duration(30.0)){
 
     listener = make_unique<tf2_ros::TransformListener>(tfBuffer);
 
-    posePub = ros::Publisher(
+    robotPosePub = ros::Publisher(
           nh.advertise<geometry_msgs::PoseWithCovarianceStamped>("/fiducial_pose", 1));
+    cameraPosePub = ros::Publisher(
+          nh.advertise<geometry_msgs::PoseWithCovarianceStamped>("/camera_pose", 1));
```
Comment (rohbotics) [inline fiducial_slam/src/map.cpp]: I think the topic `camera_pose` is not great. Maybe `fiducial_camera_pose` or put in int he `/fiducials` namespace.

File: fiducial_slam/src/map.cpp [hunk 2]
```diff
@@ -196,6 +199,7 @@ Map::Map(ros::NodeHandle &nh) : tfBuffer(ros::Duration(30.0)){
     nh.param<double>("future_date_transforms", future_date_transforms, 0.1);
     nh.param<bool>("publish_6dof_pose", publish_6dof_pose, false);
     nh.param<bool>("sum_error_in_quadrature", sum_error_in_quadrature, false);
+    nh.param<bool>("read_only_map", readOnly, false);
```
Comment (rohbotics) [inline fiducial_slam/src/map.cpp:202]: :+1:

**Review by rohbotics — APPROVED** [review]
Only 1 minor piece of feedback.

**Comment by rohbotics** [comment]
Also fit_plane needs to be chmod +x

File: fiducial_slam/src/map.cpp (hunk 1 above)
Comment (jim-v) [reply fiducial_slam/src/map.cpp]: Good point.

**Comment by rohbotics** [comment]
Good to merge

**Pull request merged**